src/validation_workflow/docker/validation_docker.py

```diff
@@ -0,0 +1,203 @@
+
+from system.temporary_directory import TemporaryDirectory
+from validation_workflow.api_request import ApiTest
+from validation_workflow.api_test_cases import TestCases
+from validation_workflow.docker.inspect_docker_image import InspectDockerImage
+from validation_workflow.validation import Validation
+from validation_workflow.validation_args import ValidationArgs
+
+
+class ValidateDocker(Validation):
+
+    def __init__(self, args: ValidationArgs) -> None:
+        super().__init__(args)
+
+    def download_artifacts(self) -> bool:
+
+        # Check if the docker daemon is running; or else, bring it up
+        if (self.is_container_daemon_running() == 0):
+            logging.info('docker_daemon is running')
+
+            # STEP 1 . pull the images for OS and OSD
+            _OS_image_name = 'opensearchproject/opensearch' if self.args.dockersource == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
+            _OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.dockersource == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
+
+            self.local_image_OS_id = self.get_image_id(_OS_image_name, self.args.version)
+            self.local_image_OSD_id = self.get_image_id(_OSD_image_name, self.args.version)
+            logging.info('the OS image ID is : ' + self.local_image_OS_id)
+            logging.info('the OSD image ID is : ' + self.local_image_OSD_id + '\n\n')
+
+            return True
+
+        else:
+            raise Exception('docker_daemon check is fail. Exiting the validation.')
+
+    def installation(self) -> bool:
+        ''' Pass this method for docker as no installation required in docker'''
+        return True
+
+    def start_cluster(self) -> bool:
+        ''' Pass this method for docker and combine it with the following method because we want to Pass the digest validation in docker first before spinning up the docker container'''
+        return True
+
+    def validation(self) -> bool:
+        # STEP 2 . inspect image digest betwwen opensearchproject(downloaed/local) and opensearchstaging(dockerHub)
+
+        self._OS_inspect_digest = InspectDockerImage(self.local_image_OS_id, self.args.OS_image, self.args.version)
+        self._OSD_inspect_digest = InspectDockerImage(self.local_image_OSD_id, self.args.OSD_image, self.args.version)
+        if self._OS_inspect_digest.inspect_digest() and self._OSD_inspect_digest.inspect_digest():
+            logging.info('Image digest is validated\n\n')
+
+            # STEP 3 . spin-up OS/OSD cluster
+            return_code, self._target_yml_file = self.run_container(
+                self.args.OS_image,
+                self.args.OSD_image,
+                self.args.version,
+                'opensearch-node1-test',
+                'opensearch-node2-test',
+                'opensearch-dashboards-test',
+            )
+            if (return_code):
+                logging.info('Cluster is running now')
```

Comment:
Why not put this under `start_cluter` method as that is actually what you are doing here?

Reply:
Thanks for the question, @gaiksaya. The reason is because we want to validate the digest before spining up the cluster,; so, I had to pass the 'start_cluster' and have both of them in the 'validation'. Hope that is fine.

Reply:
Does the order matter? I believe you can declare and call `start_cluster` anytime right??

Reply:
Looks like it does, @gaiksaya ..  we set the following at validation.py
```
          try:
            return self.download_artifacts() and self.installation() and self.start_cluster() and self.validation() and self.cleanup()
```
so, it goes to the next method only if the curent returns True.  :(